src/main/java/ui/components/FilterTextField.java

```diff
@@ -87,22 +88,39 @@ private void setup() {
                     cancel.run();
                 } else {
                     revertEdit();
-                    selectAll();
                 }
             }
         });
     }
 
-    private void performCompletion(KeyEvent e) {
-        String word = getCurrentWord() + e.getCharacter();
+    /**
+     * Completion is only started when there's effectively nothing (only whitespace)
+     * after the caret, or if there is selected text (indicating either that we are in
+     * the midst of completion, or that the user does not want the selected text and is
+     * typing to replace it)
+     * @return true if completion should be started
+     */
+    private boolean shouldStartCompletion() {
+        return getCharAfterCaret().trim().isEmpty() || !getSelectedText().isEmpty();
```

Comment:
With `ignee` selected, if you type `i`, the result is `assi[gnee]:me` (selection represented with square brackets). Typing anything else (say, `a`) results in `assa:me`. Completion is performed, but only if the current word + currently-typed letter is a prefix of some keyword.

The main reason for this change was to [not have completion cancelled as the user continues typing](https://github.com/HubTurbo/HubTurbo/blob/1167-space-key/src/test/java/guitests/FilterTextFieldTest.java#L22-L32). The other scenario of typing with some text already selected is a side effect.